control-plane/api-gateway/gatekeeper/deployment.go

```diff
@@ -69,13 +72,12 @@ func (g *Gatekeeper) deleteDeployment(ctx context.Context) error {
 func (g *Gatekeeper) deployment() *appsv1.Deployment {
 	return &appsv1.Deployment{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        g.Gateway.Name,
-			Namespace:   g.Gateway.Namespace,
-			Labels:      apigateway.LabelsForGateway(&g.Gateway),
-			Annotations: g.HelmConfig.CopyAnnotations,
+			Name:      g.Gateway.Name,
+			Namespace: g.Gateway.Namespace,
+			Labels:    apigateway.LabelsForGateway(&g.Gateway),
 		},
 		Spec: appsv1.DeploymentSpec{
-			Replicas: &g.HelmConfig.Replicas,
+			Replicas: g.GatewayClassConfig.Spec.DeploymentSpec.DefaultInstances,
```

Comment:
Just as a note, we'll want to use some of the pinning calculations so that we allow for this to range between min and max values passed in, and make it based off of the current number of replicas in the deployment. That doesn't need to go in this PR as I have some of that work in flight already, but just to describe the way it works, the logic is something like:

```go
if spec.MaxInstances != nil && currentReplicas > *spec.MaxInstances {
  replicas = *spec.MaxInstances
} else if spec.MinInstances != nil && currentReplicas < *spec.MinInstances {
  replicas = *spec.MinInstances
} else {
  replicas = firstNonNilOrZeroValueOf(currentReplicas, spec.DefaultInstances, 1)
}
```